src/averageAges.js

```diff
@@ -55,7 +75,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let averageAgeDiff = 0;
+  let children = people.filter(
+    child => people.find(mom => (mom.name === child.mother))
+  );
+
+  children = onlyWithSon
+    ? children.filter(child => child.sex === 'm')
+    : children;
+
+  const womenAges = children
+    .map(child => (
+      child.born - (people.find(mom => mom.name === child.mother).born)
```

Comment:
It is much better to use a separate variable here for `mother` reference.
It would make your code more readable.